dask_cuda/device_host_file.py

```diff
@@ -82,14 +127,9 @@ def __setitem__(self, key, value):
 
     def __getitem__(self, key):
         if key in self.host_buffer:
-            obj = self.host_buffer[key]
-            del self.host_buffer[key]
-            self.device_buffer[key] = _deserialize_if_device(obj)
-
-        if key in self.device_buffer:
-            return self.device_buffer[key]
+            return self.host_buffer[key]
```

Comment:
Just to make sure, if I put a device-backed object into a DeviceHostDisk and it gets moved to the host for space reasons, will that key show up in `self.host_buffer`?

Reply:
Yes, `self.host_buffer` is the de-facto `self.device_buffer.slow`.

Reply:
In that case shouldn't we check the device Buffer first so that it pulls the Serialized object currently residing in `self.host_buffer` back down to the device and then return that deserialized object?  I'm naively proposing switching the order of conditions here:

```python
if key in self.device_buffer:
    return self.device_buffer[key]
if key in self.host_buffer:
    return self.host_buffer[key]
raise KeyError(key)
```